test/test_submesh.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2024 Jørgen S. Dokken
+#
+# This file is part of FFCx. (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+
+from __future__ import annotations
+
+import basix.ufl
+import numpy as np
+import pytest
+import ufl
+
+import ffcx.codegeneration.jit
+from ffcx.codegeneration.utils import dtype_to_c_type, dtype_to_scalar_dtype
+
+
+def compute_tensor(forms: list[ufl.form.Form], dtype: str, compile_args: list[str]):
+    """Helper-function to compute matrix for a P1-Lagrange problem"""
+    compiled_forms, module, _ = ffcx.codegeneration.jit.compile_forms(
+        forms, options={"scalar_type": dtype}, cffi_extra_compile_args=compile_args
+    )
+
+    ffi = module.ffi
+    form0 = compiled_forms[0]
+    offsets = form0.form_integral_offsets
+    cell = module.lib.cell
+    assert offsets[cell + 1] - offsets[cell] == 1
+    integral_id = form0.form_integral_ids[offsets[cell]]
+    assert integral_id == -1
+
+    default_integral = form0.form_integrals[offsets[cell]]
+
+    A = np.zeros((3, 3), dtype=dtype)
+    w = np.array([], dtype=dtype)
+    c = np.array([], dtype=dtype)
+
+    xdtype = dtype_to_scalar_dtype(dtype)
+    coords = np.array([[1.0, 2.0, 0.0], [1.5, 2.3, 0.0], [6.0, 1.8, 0.0]], dtype=xdtype)
+
+    c_type, c_xtype = dtype_to_c_type(dtype), dtype_to_c_type(xdtype)
+    kernel = getattr(default_integral, f"tabulate_tensor_{dtype}")
+    kernel(
+        ffi.cast(f"{c_type} *", A.ctypes.data),
+        ffi.cast(f"{c_type} *", w.ctypes.data),
+        ffi.cast(f"{c_type} *", c.ctypes.data),
+        ffi.cast(f"{c_xtype} *", coords.ctypes.data),
+        ffi.NULL,
+        ffi.NULL,
+    )
+    return A
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        "float64",
+        "complex128",
+    ],
+)
+def test_submesh_codim0(dtype, compile_args):
```

Comment:
Am I right in thinking that this issue is caused when two different `ufl.Mesh`es are used in a form that have the same coordinate element? If so, maybe this test and file should be renamed to better reflect this? To me it seems like it's testing we don't get code duplication, rather than testing submeshes

Reply:
The only case when you will have two `ufl.Mesh` in the same form is whenever we are using sub-meshes.

Reply:
That's probably true, but if the user happened to have done something like

```
V = FunctionSpace(ufl.Mesh(coordinate_element), V_element)
W = FunctionSpace(ufl.Mesh(coordinate_element), W_element)
``` 

then we would still wan't that to work, right?